core/primitives/src/runtime/config_store.rs

```diff
@@ -101,25 +114,59 @@ mod tests {
     const GENESIS_PROTOCOL_VERSION: ProtocolVersion = 29;
     const RECEIPTS_DEPTH: u64 = 63;
 
-    fn check_config(protocol_version: ProtocolVersion, config_bytes: &[u8]) {
+    static OLD_CONFIGS: &[(ProtocolVersion, &str)] = &[
+        (0, include_config!("legacy_configs/29.json")),
+        (42, include_config!("legacy_configs/42.json")),
+        (48, include_config!("legacy_configs/48.json")),
+        (49, include_config!("legacy_configs/49.json")),
+        (50, include_config!("legacy_configs/50.json")),
+        (52, include_config!("legacy_configs/52.json")),
+        (53, include_config!("legacy_configs/53.json")),
+    ];
+
+    fn check_config(protocol_version: ProtocolVersion, config_bytes: &str) {
         assert_eq!(
             RuntimeConfigStore::new(None).get_config(protocol_version).as_ref(),
-            &serde_json::from_slice::<RuntimeConfig>(config_bytes).unwrap()
+            &serde_json::from_str::<RuntimeConfig>(config_bytes).unwrap()
         );
     }
 
     #[test]
     fn test_get_config() {
-        check_config(0, CONFIGS[0].1);
-        check_config(GENESIS_PROTOCOL_VERSION - 1, CONFIGS[0].1);
-        check_config(GENESIS_PROTOCOL_VERSION, CONFIGS[0].1);
+        check_config(0, OLD_CONFIGS[0].1);
+        check_config(GENESIS_PROTOCOL_VERSION - 1, OLD_CONFIGS[0].1);
+        check_config(GENESIS_PROTOCOL_VERSION, OLD_CONFIGS[0].1);
         // First non-trivial version for which runtime config was updated.
-        check_config(LowerStorageCost.protocol_version(), CONFIGS[1].1);
-        check_config(ProtocolVersion::MAX, CONFIGS.last().unwrap().1);
+        check_config(LowerStorageCost.protocol_version(), OLD_CONFIGS[1].1);
+        check_config(ProtocolVersion::MAX, OLD_CONFIGS.last().unwrap().1);
     }
 
     #[test]
     fn test_runtime_config_data() {
+        let expected_hashes = vec![
+            "2ysK1QuHLW958qxjfpnNJ9yHicncyyRrczoxtQwrkJ8o",
+            "3Rdnrr1DFxXWeQ7knpHqxnRrNUys9Ke7SUEJ2axjXvVS",
+            "61SPiLLGhNYySP2djfL1Je5LogVTAjroYnAnmzDofkJS",
+            "6cDB3NjHsG3Vq31Yp5of2hxA1FwR6xm6pb3Mc5Mw4xvK",
+            "6TWAHgsm17JdJxVjJQ9TYJERCw9rgDdzzXvopWE5XzoQ",
+            "72BDBPsrXeg23viBMTSekigRxz4jyHNxrEct2VLa1xDc",
+            "CCqfbxH12gAp9VL419FqhZ4JNxhtdimi5ETQ22mmKERH",
+        ];
+        let actual_hashes = std::iter::once(&(0, BASE_CONFIG))
+            .chain(CONFIG_DIFFS)
+            .map(|(_protocol_version, config_str)| to_base(&hash(config_str.as_bytes())))
+            .collect::<Vec<_>>();
+        assert_eq!(
+            expected_hashes, actual_hashes,
+            "\n
+Config hashes changed. \n
+If you add a new config diff, add a missing hash to the end of `expected_hashes` array.
+"
+        )
+    }
+
+    #[test]
+    fn test_old_runtime_config_data() {
```

Comment:
Nice assuarance that we didn't change values in the process of migration! Let's remove it once the PR is merged (that is, I think the only value of this test is to check this specific PR, and, once that it, wec an remove the test without the loss of confidence? )!